src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -770,15 +776,12 @@ private string GenerateExtendedAttributeName()
             string? dirName = Path.GetDirectoryName(_name);
             dirName = string.IsNullOrEmpty(dirName) ? "." : dirName;
 
-            int processId = Environment.ProcessId;
-
             string? fileName = Path.GetFileName(_name);
             fileName = string.IsNullOrEmpty(fileName) ? "." : fileName;
 
-            string trailingSeparator = (_typeFlag is TarEntryType.Directory or TarEntryType.DirectoryList) ?
-                $"{Path.DirectorySeparatorChar}" : string.Empty;
-
-            return string.Format(PaxHeadersFormat, dirName, processId, fileName, trailingSeparator);
+            return _typeFlag is TarEntryType.Directory or TarEntryType.DirectoryList ?
+                $"{dirName}/PaxHeaders.{Environment.ProcessId}/{fileName}" :
+                $"{dirName}/PaxHeaders.{Environment.ProcessId}/{fileName}{Path.DirectorySeparatorChar}";
```

Comment:
Oops. this condition is reversed... but no tests are failing as a result.  I'll fix the inversion,  but it seems like a test gap that should be addressed.

Reply:
I can add a test for that.